syntax = "proto3";

message IChatMessage {
    int32 message_id = 1;
    int32 sender_id = 2;
    string sender_name = 3;
    string message_content = 4;
    string message_timestamp = 5;
}

message IChat {
    int32 chat_id = 1;
    string interlocutor_name = 2; // имя собеседника - название чата
    repeated IChatMessage messages = 3;
}

message IGetResponse {
    repeated IChat chats = 1;
    bool has_more_chats = 2; // флаг, указывающий что есть еще чаты для подгрузки
}
